scripts/Crawlers/BaseCrawler.py

```diff
@@ -0,0 +1,209 @@
+from datalad import api
+import git
+import os
+import re
+import requests
+import abc
+import json
+
+
+class BaseCrawler:
+    def __init__(self, github_token, config_path, verbose, force):
+        self.repo = git.Repo()
+        self.username = self._check_requirements()
+        self.github_token = github_token
+        self.config_path = config_path
+        self.verbose = verbose
+        self.force = force
+        self.git = git
+        self.datalad = api
+
+    @abc.abstractmethod
+    def get_all_dataset_metadata(self):
+        """
+        Get relevant datasets' metadata from platform.
+
+        Retrieves datasets' metadata that needs to be in CONP-datasets
+        from platform specific to each crawler like Zenodo, OSF, etc.
```

Comment:
Explain what are the constraints on the descriptions. They have to be JSON objects, with all required properties to build a DATS file. Title has to be unique. Explain that this will DATS fields and that the crawler can always add more as long as it complies to the DATS schema (add link to it).